hat/app/org/hatdex/hat/api/controllers/RichData.scala

```diff
@@ -24,24 +24,48 @@
 
 package org.hatdex.hat.api.controllers
```

Comment:
Suggest moving all of the contract-related logic into a separate controller under separate module